src/e2e-test/java/io/cdap/plugin/http/stepsdesign/HttpPlugin.java

```diff
@@ -0,0 +1,395 @@
+        , countRecords);
+    } else {
+      Assert.assertEquals(0, countRecords);
+    }
+  }
+
+  @When("Source is GCS bucket")
+  public void sourceIsGCSBucket() throws InterruptedException {
+    CdfStudioActions.selectGCS();
+  }
+
+  @Then("Enter the GCS Properties with {string} GCS bucket")
+  public void enterTheGCSPropertiesWithGCSBucket(String bucket) throws InterruptedException, IOException {
+    CdfGcsActions.gcsProperties();
+    CdfGcsActions.enterReferenceName();
+    CdfGcsActions.enterProjectId();
+    CdfGcsActions.getGcsBucket(E2ETestUtils.pluginProp(bucket));
+    CdfGcsActions.selectFormat("csv");
+    CdfGcsActions.skipHeader();
+    CdfGcsActions.getSchema();
+    SeleniumHelper.waitElementIsVisible(HttpPluginLocators.getSchemaLoadComplete, 5);
+  }
+
+  @Then("Close the GCS Properties")
+  public void closeTheGCSProperties() {
+    CdfGcsActions.closeButton();
+  }
+
+  @Then("Link Source GCS and Sink HTTP to establish connection")
+  public void linkSourceGCSAndSinkHTTPToEstablishConnection() {
+    SeleniumHelper.waitElementIsVisible(HttpPluginLocators.toHTTP);
+    SeleniumHelper.dragAndDrop(CdfStudioLocators.fromGCS, HttpPluginLocators.toHTTP);
+  }
+
+  @Then("Enter the HTTP Sink Properties")
+  public void enterTheHTTPSinkProperties() {
+    HttpPluginActions.enterReferenceName(E2ETestUtils.pluginProp("httpSinkReferenceName"));
+    HttpPluginActions.enterHttpUrl(E2ETestUtils.pluginProp("httpSinkUrl"));
+    HttpPluginActions.selectHttpMethodSink(E2ETestUtils.pluginProp("httpSinkMethod"));
+    HttpPluginActions.selectSinkMessageFormat(E2ETestUtils.pluginProp("httpSinkMessageFormat"));
+    HttpPluginActions.clickValidateButton();
+  }
+
+  @Then("Validate OUT record count is equal to IN record count")
+  public void validateOUTRecordCountIsEqualToINRecordCount() {
+    Assert.assertEquals(recordOut(), recordIn());
+  }
+
+  @Then("Capture output schema")
+  public void captureOutputSchema() {
+    SeleniumHelper.waitElementIsVisible(SeleniumDriver.getDriver().findElement(
+      By.xpath("//div[@data-cy='schema-fields-list']//*[@placeholder='Field name']")), 10L);
+    List<WebElement> propertiesOutputSchemaElements = SeleniumDriver.getDriver().findElements(
+      By.xpath("//div[@data-cy='schema-fields-list']//*[@placeholder='Field name']"));
+    for (WebElement element : propertiesOutputSchemaElements) {
+      propertiesSchemaColumnList.add(element.getAttribute("value"));
+    }
+  }
+
+  @Then("Click on PreviewData for GCS")
+  public void clickOnPreviewDataForGcs() {
+    CdfGcsActions.clickPreviewData();
+  }
+
+  @Then("Click on PreviewData for HTTP")
+  public void clickOnPreviewDataForHttp() {
+    HttpPluginActions.clickHttpPreviewData();
+  }
+
+  @Then("Verify Preview output schema matches the outputSchema captured in properties")
+  public void verifyPreviewOutputSchemaMatchesTheOutputSchemaCapturedInProperties() {
+    List<String> previewSchemaColumnList = new ArrayList<>();
```

Comment:
schema contains field name, type and nullability. Please check all of these fields